pkg/apis/apps/validation/validation.go

```diff
@@ -321,6 +321,7 @@ func ValidateDaemonSetUpdate(ds, oldDS *apps.DaemonSet, opts apivalidation.PodVa
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&ds.ObjectMeta, &oldDS.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidateDaemonSetSpecUpdate(&ds.Spec, &oldDS.Spec, field.NewPath("spec"))...)
 	allErrs = append(allErrs, ValidateDaemonSetSpec(&ds.Spec, field.NewPath("spec"), opts)...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(ds.Spec.Selector, oldDS.Spec.Selector, field.NewPath("spec").Child("selector"))...)
```

Comment:
There's `ValidateDaemonSetSpecUpdate` invoked two lines above and this should be part of that method.

Reply:
Ah, thank you for point that out, I'll move the code in few hours.